### ISSUE TITLE
Shellcheck version on CI is __old__

### ISSUE THREAD

**Issue by jbertran** [scality/metalk8s#699]
Bare Centos 7:

```
ShellCheck - shell script analysis tool
version: 0.3.5
```

Average dev machine:

```
ShellCheck - shell script analysis tool
version: 0.6.0
```

Result: developers pick up errors that the CI doesn't. The fix is either to build it locally, install it using Cabal, or compile it statically and drop it into the worker.

**Comment by jbertran** [comment]
On a CentOS box, 

```
yum install -y epel-release
yum install -y cabal-install
cabal update && cabal install shellcheck
```

This crashes and burns after 5 minutes of install time. Binary requires stuffing data from a random googleapi URL in an executable path (I'd accept a github release URL). This needs to get fixed but for now the CI is _less_ restrictive than dev machines, so I'm content letting it be. Leaving this for tracking and reference when spot-fixing lint errors from recent versions of the tool.

**Comment by NicolasT** [comment]
Was about to suggest "Take the Fedora RPM and install it as-is, most Haskell software is statically linked anyway", but that won't work because the binary is linked against `libm` at symbol version `GLIBC_2.27`, which is not what CentOS ships with.

However, you could consider using a Docker worker for the shell lint-step which is based on Fedora and has the package installed in it? In the end, this is not a worker we use to test the actual software we ship, but lint it (purely static analysis).

**Comment by jbertran** [comment]
> However, you could consider using a Docker worker for the shell lint-step which is based on Fedora and has the package installed in it? In the end, this is not a worker we use to test the actual software we ship, but lint it (purely static analysis).

It's an option as well, I don't feel good about the fact that I need to mix and match base images for CI though.

**Comment by jbertran** [comment]
Closed by #700
